base_layer/core/src/transactions/types.rs

```diff
@@ -102,3 +102,13 @@ impl CryptoFactories {
         }
     }
 }
+
+/// Uses Arc's internally so calling clone on this is cheap, no need to wrap this in an Arc
```

Comment:
How about `#[derive(Clone)]` on the struct?

Reply:
I prefer a manual impl as this will make you think about what happens here. If you add a non arc object to cryptofactories you might want to make each implementation of it an arc again.

Reply:
sdbondi is right as usual, but I feel the same as SW here.